File: 1037-minimum-number-of-k-consecutive-bit-flips/minimum-number-of-k-consecutive-bit-flips.py
class Solution:
    def minKBitFlips(self, nums: List[int], k: int) -> int:
        n = len(nums)
        flipped, res, isFlipped = 0, 0, [0] * n

        for i in range(n):
            if i >= k:
                flipped ^= isFlipped[i-k]
            if flipped == nums[i]:
                if i + k > n:
                    return -1
                isFlipped[i] = 1
                flipped ^= 1
                res += 1
        
        return res

File: 1037-minimum-number-of-k-consecutive-bit-flips/minimum-number-of-k-consecutive-bit-flips.java
class Solution {
    public int minKBitFlips(int[] nums, int k) {
        int n = nums.length, flipped = 0, res = 0;
        int[] isFlipped = new int[n];

        for (int i = 0; i < n; i++) {
            if (i >= k) {
                flipped ^= isFlipped[i-k];
            }
            if (flipped == nums[i]) {
                if (i + k > n ) {
                    return -1;
                }
                isFlipped[i] = 1;
                flipped ^= 1;
                res += 1;
            }
        }
        return res;
    }
}